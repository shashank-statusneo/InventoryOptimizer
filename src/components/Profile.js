import React, { useEffect, useState } from 'react'
import Box from '@mui/material/Box'
import Card from '@mui/material/Card'
import CardActions from '@mui/material/CardActions'
import CardContent from '@mui/material/CardContent'
import Button from '@mui/material/Button'
import Typography from '@mui/material/Typography'
import { CardMedia } from '@mui/material'
import Grid from '@mui/material'
import UserSession from '../services/auth'

const Profile = () => {

    const [user, setUser] = useState({
        first_name: 'Unknown',
        last_name: 'User',
        email: 'N/A',
    })

    useEffect(() => {
        setUser(UserSession.getUser())
        console.log(user)
    }, [])
    return (
        <Card
            sx={{ maxWidth: 300 }}
            style={{
                justify: 'center',
                alignContent: 'center',
                alignItems: 'center',
            }}
        >
            <CardContent>
                <CardMedia
                    sx={{ height: 140 }}
                    image="./profilepic.svg"
                    title="profile pic"
                />
                <Typography variant="h5" component="div">
                    {user.first_name} {user.last_name}
                </Typography>
                <Typography sx={{ mb: 1.5 }} color="text.secondary">
                    {user.email}
                </Typography>
            </CardContent>
        </Card>
    )
}

export default Profile
