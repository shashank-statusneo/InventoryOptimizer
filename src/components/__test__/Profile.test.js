import { render, screen, cleanup } from "@testing-library/react";
import Profile from "../Profile";
import renderer from 'react-test-renderer'


test("test", () => {
    expect(true).toBe(true)
})

afterEach( () => {
    cleanup()
})

it('snapshot <Profile />', () => {
    const tree = renderer.create(<Profile  />).toJSON()
    expect(tree).toMatchSnapshot()
})