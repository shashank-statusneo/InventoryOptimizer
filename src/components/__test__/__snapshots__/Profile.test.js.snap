// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot <Profile /> 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1xez8po-MuiPaper-root-MuiCard-root"
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "justify": "center",
    }
  }
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiCardMedia-root css-1999db7-MuiCardMedia-root"
      role="img"
      style={
        Object {
          "backgroundImage": "url(\\"./profilepic.svg\\")",
        }
      }
      title="profile pic"
    />
    <div
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Unknown
       
      User
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-9eg9l8-MuiTypography-root"
    >
      N/A
    </p>
  </div>
</div>
`;
